Caused by: java.lang.NullPointerException: Attempt to get length of null array

## Hockey app:
4.1.1, 4.0.1

## Android: 
6.0.1

## Manufacturer: 
samsung

## Model: 
SM-N910W8

## Exception:
```
java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$3.done(AsyncTask.java:309)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
	at java.util.concurrent.FutureTask.run(FutureTask.java:242)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
	at java.lang.Thread.run(Thread.java:818)
Caused by: java.lang.NullPointerException: Attempt to get length of null array
	at net.hockeyapp.android.metrics.Persistence.isFreeSpaceAvailable(Persistence.java:248)
	at net.hockeyapp.android.metrics.Persistence.persist(Persistence.java:80)
	at net.hockeyapp.android.metrics.Channel.synchronize(Channel.java:91)
	at net.hockeyapp.android.metrics.Channel.enqueue(Channel.java:72)
	at net.hockeyapp.android.metrics.Channel.enqueueData(Channel.java:141)
	at net.hockeyapp.android.metrics.MetricsManager$1.doInBackground(MetricsManager.java:340)
	at net.hockeyapp.android.metrics.MetricsManager$1.doInBackground(MetricsManager.java:334)
	at android.os.AsyncTask$2.call(AsyncTask.java:295)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	... 3 more
```

--- a/hockeysdk/src/main/java/net/hockeyapp/android/metrics/Persistence.java
+++ b/hockeysdk/src/main/java/net/hockeyapp/android/metrics/Persistence.java
@@ -269,12 +269,13 @@ class Persistence {
         // TODO Check for available disk space as well.
         synchronized (LOCK) {
             Context context = getContext();
-            if ((context.getFilesDir()) != null) {
+            if (context.getFilesDir() != null) {
                 File filesDir = context.getFilesDir();
                 String path = filesDir.getAbsolutePath() + BIT_TELEMETRY_DIRECTORY;
                 if (!TextUtils.isEmpty(path)) {
                     File dir = new File(path);
-                    return (dir.listFiles().length < MAX_FILE_COUNT);
+                    File[] files = dir.listFiles();
+                    return files != null && files.length < MAX_FILE_COUNT;
                 }
             }
             return false;

